### DISCUSSION TITLE
variable "using" is not a type name

### DISCUSSION THREAD

**Discussion by clock999** [microsoft/vscode-cpptools#8278] [General]
HI,

The vscode report the error message "variable "using" is not a type name" with below code in the cpp file.
```
using namespace std;
```
The config file is as below.
```
{
    "configurations": [
        {
            "name": "Linux",
            "includePath": [
                "${workspaceFolder}/**",
                "/usr/include/c++/v1"
            ],
            "defines": [],
            "compilerPath": "/usr/bin/g++",
            "cppStandard": "c++11",
            "intelliSenseMode": "linux-gcc-x64"
        }
    ],
    "version": 4
}
```
Could you help me with this problem? Thanks a lot!

**Comment by sean-mcmanus** [comment]
What is the error code number? Can you try removing the /usr/include/c++/v1 -- that could break the system include path ordering.

**Reply by clock999** [reply]
Hello,

If I remove /usr/include/c++/v1, there will be an including error "cannot open source file "string"".
How do check the error code number? I copied the whole line of error message below. Maybe the error code is 757?
variable "using" is not a type nameC/C++(757)

**Comment by clock999** [comment]
Hello,

If I remove  /usr/include/c++/v1, there will be an including error "cannot open source file "string"".
How do check the error code number? I copied the whole line of error message below. Maybe the error code is 757?
variable "using" is not a type nameC/C++(757)

**Reply by sean-mcmanus** [reply]
The path to string is supposed to come from the /usr/bin/g++ compilerPath. Do you have g++ installed? If you set C_Cpp.loggingLevel to "Debug" do you see any error regarding querying your compiler? What OS are you using?
